p2p/ipld/read_test.go

```diff
@@ -153,6 +156,88 @@ func TestGetLeafData(t *testing.T) {
 	}
 }
 
+func TestBlockRecovery(t *testing.T) {
+	// adjustedLeafSize describes the size of a leaf that will not get split
+	adjustedLeafSize := types.MsgShareSize
+
+	originalSquareWidth := 2
+	sharecount := originalSquareWidth * originalSquareWidth
+	extendedSquareWidth := originalSquareWidth * originalSquareWidth
+	extendedShareCount := extendedSquareWidth * extendedSquareWidth
+
+	// generate test data
+	quarterShares := generateRandNamespacedRawData(sharecount, types.NamespaceSize, adjustedLeafSize)
+	allShares := generateRandNamespacedRawData(sharecount, types.NamespaceSize, adjustedLeafSize)
+
+	testCases := []struct {
+		name string
+		// blockData types.Data
+		shares    [][]byte
+		expectErr bool
+		errString string
+		d         int // number of shares to delete
+	}{
+		// missing more shares causes RepairExtendedDataSquare to hang see
+		// https://github.com/lazyledger/rsmt2d/issues/21
+		{"missing 1/4 shares", quarterShares, false, "", extendedShareCount / 4},
+		{"missing all but one shares", allShares, true, "failed to solve data square", extendedShareCount - 1},
```

Comment:
We have some work to do on `RepairExtendedDataSquare` [#21](https://github.com/lazyledger/rsmt2d/issues/21)

Reply:
Ouch, that issue sounds weird 🤔 Is it enough to reconsider choices as per https://github.com/lazyledger/lazyledger-core/issues/230#issuecomment-808249600 ?

Reply:
hmmm... for the time being, no, but in the near future, yes. once we have a working version of ADR 002, we're going to have to fix [#21](https://github.com/lazyledger/rsmt2d/issues/21) somehow, and `RepairExtendedDataSquare` could use a fresh coat of paint, so we might want to pursue option 3 then.